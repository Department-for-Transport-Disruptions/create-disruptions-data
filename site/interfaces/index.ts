import { NextApiRequest } from "next";
import { z } from "zod";
import { ServerResponse } from "http";
import { Service, Stop } from "../schemas/consequence.schema";
import { ConsequenceType } from "../schemas/type-of-consequence.schema";

export interface ErrorInfo {
    errorMessage: string;
    id: string;
    userInput?: string;
}

export interface ResponseWithLocals extends ServerResponse {
    locals: {
        nonce: string;
        csrfToken: string;
    };
}

export interface CookiePolicy {
    essential: boolean;
    usage: boolean;
}

export interface CookiesApiRequest extends NextApiRequest {
    body: {
        tracking: "on" | "off";
    };
}

export interface FormBase<T> {
    value?: string;
    display: string;
    displaySize?: "s" | "m" | "l" | "xl";
    inputName: Extract<keyof T, string>;
    initialErrors?: ErrorInfo[];
    stateUpdater: (change: string, field: keyof T) => void;
    schema?: z.ZodTypeAny;
}

export interface DisplayValuePair<T = string> {
    display: string;
    value: T;
    checked?: boolean;
}

export interface Feedback {
    question: string;
    answer: string;
}

export interface TestInputs {
    field1: string;
    field2: boolean;
    field3: string;
    field4: string;
}

export interface PageState<T> {
    errors: ErrorInfo[];
    inputs: T;
}

export interface SocialMediaPost {
    messageToAppear: string;
    publishDate: string;
    publishTime: string;
    accountToPublish: string;
}

export interface CreateConsequenceProps<T> {
    initialPageState: PageState<Partial<T>>;
    previousConsequenceInformation: ConsequenceType;
    initialServices?: Service[];
    initialStops?: Stop[];
    allStops?: Stop[]
}
