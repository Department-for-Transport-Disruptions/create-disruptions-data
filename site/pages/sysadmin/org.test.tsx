import { render } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import renderer from "react-test-renderer";
import { afterEach, describe, expect, it, vi } from "vitest";
import { PageState } from "../../interfaces";
import ManageOrgs, { ManageOrgProps } from "./org.page";

const blankInputs: PageState<Partial<ManageOrgProps>> = {
    inputs: {},
    errors: [],
};

const withInputs: PageState<Partial<ManageOrgProps>> = {
    inputs: {
        name: "test-org",
        adminAreas: [
            { administrativeAreaCode: "001", name: "Area 1", shortName: "A1" },
            { administrativeAreaCode: "002", name: "Area 2", shortName: "A2" },
            { administrativeAreaCode: "051", name: "Area 51", shortName: "A51" },
        ],
        orgAdminAreas: [
            { administrativeAreaCode: "001", name: "Area 1", shortName: "A1" },
            { administrativeAreaCode: "002", name: "Area 2", shortName: "A2" },
        ],
    },
    errors: [],
};
describe("manageOrgs", () => {
    vi.mock("../../data/dynamo", () => ({
        getOrganisationInfoById: vi.fn(),
    }));

    vi.mock("@create-disruptions-data/shared-ts/utils/refDataApi", () => ({
        fetchAdminAreaCodes: vi.fn(),
    }));

    afterEach(() => {
        vi.resetAllMocks();
    });

    it("should render correctly when there are no inputs", () => {
        const tree = renderer.create(<ManageOrgs {...blankInputs} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should render correctly with inputs", () => {
        const tree = renderer.create(<ManageOrgs {...withInputs} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should add admin area to list when selected", async () => {
        const { unmount, getByLabelText, getByText, getAllByRole } = render(<ManageOrgs {...withInputs} />);

        await userEvent.click(getByLabelText("NaPTAN AdminArea"));
        await userEvent.click(getByText("051 - Area 51"));

        const expectedRow = getAllByRole("row")[3];

        expect(expectedRow.innerHTML).toContain("051 - Area 51");

        unmount();
    });
});
