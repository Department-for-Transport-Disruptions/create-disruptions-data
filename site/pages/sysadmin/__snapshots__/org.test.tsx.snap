// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`manageOrgs > should render correctly when there are no inputs 1`] = `
[
  <div
    className="app-phase-banner__wrapper"
  >
    <div
      className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        className="govuk-phase-banner__content"
      >
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
        <span
          className="govuk-phase-banner__text"
        >
          This is a new service â€“ your
           
          <a
            className="govuk-link"
            href="/feedback"
            id="feedback-link"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            feedback
          </a>
           
          will help us to improve it.
        </span>
      </p>
    </div>
  </div>,
  <div
    className="govuk-width-container"
  >
    <main
      className="govuk-main-wrapper"
      id="main-content"
    >
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/sysadmin/org"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <h1
              className="govuk-heading-xl"
            >
              Add new organisation
            </h1>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="name"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="name-input"
                >
                  Organisation name
                </label>
                <div
                  className=""
                >
                  <input
                    className="govuk-input w border-3"
                    disabled={false}
                    id="name-input"
                    maxLength={100}
                    name="name"
                    onChange={[Function]}
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="adminAreaCodes-input"
                >
                  NaPTAN AdminArea
                </label>
                <div
                  className=""
                >
                  <div
                    className=" css-b62m3t-container"
                    id="adminAreaCodes"
                    onKeyDown={[Function]}
                  >
                    <span
                      className="css-1f43avz-a11yText-A11yText"
                      id="react-select-dropdown-adminAreaCodes-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      className="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      className=" css-178nxlf-control"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className=" css-1fdsijx-ValueContainer"
                      >
                        <div
                          className=" css-1jqq78o-placeholder"
                          id="react-select-dropdown-adminAreaCodes-placeholder"
                        >
                          Select code
                        </div>
                        <div
                          className=" css-qbdosj-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-dropdown-adminAreaCodes-placeholder"
                            aria-expanded={false}
                            aria-haspopup={true}
                            autoCapitalize="none"
                            autoComplete="off"
                            autoCorrect="off"
                            className=""
                            disabled={false}
                            id="adminAreaCodes-input"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            role="combobox"
                            spellCheck="false"
                            style={
                              {
                                "background": 0,
                                "border": 0,
                                "color": "inherit",
                                "font": "inherit",
                                "gridArea": "1 / 2",
                                "label": "input",
                                "margin": 0,
                                "minWidth": "2px",
                                "opacity": 1,
                                "outline": 0,
                                "padding": 0,
                                "width": "100%",
                              }
                            }
                            tabIndex={0}
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        className=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          className=" css-1u9des2-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          className=" css-1xc3v61-indicatorContainer"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-tj5bde-Svg"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <table
                  className="govuk-table"
                >
                  <thead
                    className="govuk-table__head"
                  >
                    <tr
                      className="govuk-table__row"
                    />
                  </thead>
                  <tbody
                    className="govuk-table__body"
                  />
                </table>
              </div>
            </div>
            <input
              name="PK"
              type="hidden"
            />
            <input
              name="adminAreaCodes"
              type="hidden"
              value={[]}
            />
            <input
              name="mode"
              type="hidden"
            />
            <button
              className="govuk-button mt-2"
              data-module="govuk-button"
            >
              Add
            </button>
            <a
              className="govuk-button mt-2 ml-7 govuk-button--secondary"
              href="/sysadmin/manage-organisations"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              role="button"
            >
              Cancel
            </a>
          </form>
        </div>
      </div>
    </main>
    <div>
      <h2
        className="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        className="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link:
         
        <a
          className="govuk-link govuk-!-font-size-19"
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Contact us
        </a>
      </p>
    </div>
  </div>,
  <footer
    className="govuk-footer"
    role="contentinfo"
  >
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-footer__meta"
      >
        <div
          className="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            className="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            className="govuk-footer__inline-list"
          >
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Help
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/cookies"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Cookies
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Contact
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/accessibility"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Accessibility
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Terms and conditions
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/privacy"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            className="govuk-footer__body"
          >
            Built by the
             
            <a
              className="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            className="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            className="govuk-footer__licence-description"
          >
            All content is available under the
             
            <a
              className="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          className="govuk-footer__meta-item"
        >
          <a
            className="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;

exports[`manageOrgs > should render correctly with inputs 1`] = `
[
  <div
    className="app-phase-banner__wrapper"
  >
    <div
      className="govuk-phase-banner max-w-[960px] w-[90%] border-0 mx-auto app-phase-banner app-width-container"
    >
      <p
        className="govuk-phase-banner__content"
      >
        <strong
          className="govuk-tag govuk-phase-banner__content__tag"
        >
          beta
        </strong>
        <span
          className="govuk-phase-banner__text"
        >
          This is a new service â€“ your
           
          <a
            className="govuk-link"
            href="/feedback"
            id="feedback-link"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            feedback
          </a>
           
          will help us to improve it.
        </span>
      </p>
    </div>
  </div>,
  <div
    className="govuk-width-container"
  >
    <main
      className="govuk-main-wrapper"
      id="main-content"
    >
      <div
        className="govuk-grid-row"
      >
        <div
          className="govuk-grid-column-two-thirds"
        >
          <form
            action="/api/sysadmin/org"
            method="post"
          >
            <input
              name="csrf_token"
              type="hidden"
              value=""
            />
            <h1
              className="govuk-heading-xl"
            >
              Add new organisation
            </h1>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
                id="name"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="name-input"
                >
                  Organisation name
                </label>
                <div
                  className=""
                >
                  <input
                    className="govuk-input w border-3"
                    defaultValue="test-org"
                    disabled={false}
                    id="name-input"
                    maxLength={100}
                    name="name"
                    onChange={[Function]}
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              className="govuk-form-group"
            >
              <div
                className="govuk-form-group"
              >
                <label
                  className="govuk-label govuk-label--s"
                  htmlFor="adminAreaCodes-input"
                >
                  NaPTAN AdminArea
                </label>
                <div
                  className=""
                >
                  <div
                    className=" css-b62m3t-container"
                    id="adminAreaCodes"
                    onKeyDown={[Function]}
                  >
                    <span
                      className="css-1f43avz-a11yText-A11yText"
                      id="react-select-dropdown-adminAreaCodes-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      className="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      className=" css-178nxlf-control"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className=" css-1fdsijx-ValueContainer"
                      >
                        <div
                          className=" css-1jqq78o-placeholder"
                          id="react-select-dropdown-adminAreaCodes-placeholder"
                        >
                          Select code
                        </div>
                        <div
                          className=" css-qbdosj-Input"
                          data-value=""
                        >
                          <input
                            aria-autocomplete="list"
                            aria-describedby="react-select-dropdown-adminAreaCodes-placeholder"
                            aria-expanded={false}
                            aria-haspopup={true}
                            autoCapitalize="none"
                            autoComplete="off"
                            autoCorrect="off"
                            className=""
                            disabled={false}
                            id="adminAreaCodes-input"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onFocus={[Function]}
                            role="combobox"
                            spellCheck="false"
                            style={
                              {
                                "background": 0,
                                "border": 0,
                                "color": "inherit",
                                "font": "inherit",
                                "gridArea": "1 / 2",
                                "label": "input",
                                "margin": 0,
                                "minWidth": "2px",
                                "opacity": 1,
                                "outline": 0,
                                "padding": 0,
                                "width": "100%",
                              }
                            }
                            tabIndex={0}
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                      <div
                        className=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          className=" css-1u9des2-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          className=" css-1xc3v61-indicatorContainer"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-tj5bde-Svg"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <table
                  className="govuk-table"
                >
                  <thead
                    className="govuk-table__head"
                  >
                    <tr
                      className="govuk-table__row"
                    />
                  </thead>
                  <tbody
                    className="govuk-table__body"
                  >
                    <tr
                      className="govuk-table__row"
                    >
                      <td
                        className="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                      >
                        001 - Area 1
                      </td>
                      <td
                        className="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                      >
                        <button
                          className="govuk-link"
                          id="remove-stop-001"
                          onClick={[Function]}
                        >
                          Remove
                        </button>
                      </td>
                    </tr>
                    <tr
                      className="govuk-table__row"
                    >
                      <td
                        className="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                      >
                        002 - Area 2
                      </td>
                      <td
                        className="govuk-table__cell align-middle overflow-hidden px-2 max-w-2xl "
                      >
                        <button
                          className="govuk-link"
                          id="remove-stop-002"
                          onClick={[Function]}
                        >
                          Remove
                        </button>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
            <input
              name="PK"
              type="hidden"
            />
            <input
              name="adminAreaCodes"
              type="hidden"
              value={
                [
                  "001",
                  "002",
                ]
              }
            />
            <input
              name="mode"
              type="hidden"
            />
            <button
              className="govuk-button mt-2"
              data-module="govuk-button"
            >
              Add
            </button>
            <a
              className="govuk-button mt-2 ml-7 govuk-button--secondary"
              href="/sysadmin/manage-organisations"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              role="button"
            >
              Cancel
            </a>
          </form>
        </div>
      </div>
    </main>
    <div>
      <h2
        className="govuk-heading-s"
      >
        Help and Support
      </h2>
      <p
        className="govuk-body"
      >
        If you are having problems, please contact the Create Disruption Service via this link:
         
        <a
          className="govuk-link govuk-!-font-size-19"
          href="/contact"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Contact us
        </a>
      </p>
    </div>
  </div>,
  <footer
    className="govuk-footer"
    role="contentinfo"
  >
    <div
      className="govuk-width-container"
    >
      <div
        className="govuk-footer__meta"
      >
        <div
          className="govuk-footer__meta-item govuk-footer__meta-item--grow"
        >
          <h2
            className="govuk-visually-hidden"
          >
            Support links
          </h2>
          <ul
            className="govuk-footer__inline-list"
          >
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Help
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/cookies"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Cookies
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/contact"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Contact
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/accessibility"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Accessibility
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="https://www.gov.uk/help/terms-conditions"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Terms and conditions
              </a>
            </li>
            <li
              className="govuk-footer__inline-list-item"
            >
              <a
                className="govuk-footer__link"
                href="/privacy"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onTouchStart={[Function]}
              >
                Privacy
              </a>
            </li>
          </ul>
          <p
            className="govuk-footer__body"
          >
            Built by the
             
            <a
              className="govuk-footer__link"
              href="https://www.gov.uk/government/organisations/department-for-transport"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="noreferrer"
              target="_blank"
            >
              Department for Transport
            </a>
          </p>
          <svg
            aria-hidden="true"
            className="govuk-footer__licence-logo"
            focusable="false"
            height="17"
            viewBox="0 0 483.2 195.7"
            width="41"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
              fill="currentColor"
            />
          </svg>
          <span
            className="govuk-footer__licence-description"
          >
            All content is available under the
             
            <a
              className="govuk-footer__link"
              href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
              rel="license"
            >
              Open Government Licence v3.0
            </a>
            , except where otherwise stated
          </span>
        </div>
        <div
          className="govuk-footer__meta-item"
        >
          <a
            className="govuk-footer__link govuk-footer__copyright-logo bg-govuk-crest"
            href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            Â© Crown copyright
          </a>
        </div>
      </div>
    </div>
  </footer>,
]
`;
